dashboard/modules/snapshot/snapshot_schema.json

```diff
@@ -71,6 +71,41 @@
                 }
               }
             },
+            "jobSubmission": {
```

Comment:
Does this schema make sense (in particular, the choice of which fields are nullable?)

Reply:
1. do these jobs have names at all? i don't think they do, but just double checking
2. can we add an error type field here (specifically for when the status is errored?) it would be good to surface if, for example, the failure was a runtime env failure. Specifically, useful error types could be:
a. runtime env setup
b. internal system failure
c. user script error
3. what is namespace? how is it populated? i thought it's possible to have multiple sub drivers that point to different namespaces in the same ray job submission.

other than that, lgtm!

Reply:
1. I don't think so, just the Job ID.  There's no way to specify a name in the Jobs API
2. Sure, let me look into this.  You're saying to add a field to the schema in this PR, and then figure out the error plumbing in a future PR?  It seems out of the scope of this PR

3. [Namespace discussion](https://github.com/ray-project/ray/pull/22225#issuecomment-1033271264) it doesn't mean anything right now, but we'll leave it here for the future.  It'll probably be something like you set a namespace in the job submission, and then each driver reads an env var and sets its namespace to that namespace, but you can override it per driver by specifying a different namespace.

Reply:
1. No names ATM.
2. Good idea. Should this be an "error type" or just different top-level statuses though?
3. It's not very well-defined; maybe we should just drop namespace for now and we can add it later if it makes sense.

Reply:
For 2, I think I'll add error_type as a top-level field rather than increase the number of top-level statuses.  Right now with the SDK the user can do a check like `if status == FAILED`, but this type of check becomes more complicated if there are many top-level failure types.

For 3, I'll drop namespace for now.